include/Battlemines_2/board_console.h

```diff
@@ -1,11 +1,13 @@
 #include "board.h"
 #include <functional>
 
+typedef Position (*userInputPointer)(Board const &, Player , RNGPointer );
```

Comment:
#should probably rename such that it becomes more clear that it's a function pointer; i.e. `UserInputFunctionPointer`